main.c

```diff
@@ -235,20 +236,84 @@ static bool path_exists(const char *path) {
 	return path && access(path, R_OK) != -1;
 }
 
-static const char *get_output_dir(void) {
-	static const char *output_dirs[] = {
-		"GRIM_DEFAULT_DIR",
-		"XDG_PICTURES_DIR",
-	};
+char *get_xdg_pictures_dir(void) {
+	const char *home_dir = getenv("HOME");
+	if (home_dir == NULL) {
+		return NULL;
+	}
+
+	char *config_file;
+	const char *config_home = getenv("XDG_CONFIG_HOME");
+	if (config_home == NULL || config_home[0] == 0) {
+		size_t len = strlen(home_dir) + strlen("/.config/user-dirs.dirs") + 1;
```

Comment:
Can we rename this to `size`? `len` is a bit confusing since this value is in fact `strlen(config_file) + 1`, not `strlen(config_file)`.